### ISSUE TITLE
New wheels of rasterio no longer read AIG grids

### ISSUE THREAD

**Issue by johanvdw** [sgillies/rasterio-wheels#103]
## Expected behavior and actual behavior.

When trying to open a [AIG grid](https://gdal.org/drivers/raster/aig.html), the latest linux wheels for rasterio give an error. This was not present before.

## Steps to reproduce the problem.

```
import rasterio as rio
with rio.open("tests/data/ff_bt_t10_h/", driver="AIG") as dataset:
    data = dataset.read()
```
will result in
```
---------------------------------------------------------------------------
CPLE_OpenFailedError                      Traceback (most recent call last)
rasterio/_base.pyx in rasterio._base.DatasetBase.__init__()

rasterio/_base.pyx in rasterio._base.open_dataset()

rasterio/_err.pyx in rasterio._err.exc_wrap_pointer()

CPLE_OpenFailedError: 'tests/data/ff_bt_t10_h/' not recognized as a supported file format.
```

## Installation Method

Working version:  rasterio-1.3.5-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
Failing versions:
* rasterio-1.3.5.post1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
* rasterio-1.3.6-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl

I've added my testfiles: 
[aig_test.tar.gz](https://github.com/rasterio/rasterio/files/10731005/aig_test.tar.gz)

**Comment by sgillies** [comment]
Yep, GDAL changed its default driver configuration in switching to cmake. Some defaults are no longer defaults. I'll figure this out for 1.3.7.

**Comment by sgillies** [comment]
Resolved by #110
